### PR TITLE
Fix group handling for widget tests (#1249).

### PR DESCRIPTION
**Pull request by pq**
Fixes: #1249.

image

image

@stevemessick @devoncarew

### CODE REVIEW

File: src/io/flutter/run/test/FlutterTestEventsConverter.java [hunk 1]
```diff
@@ -35,9 +35,9 @@ public FlutterTestEventsConverter(@NotNull String testFrameworkName,
    * Checks if the given group is one created dynamically by flutter_test in the widget_tester
    * widget test wrapper.
    */
-  private static boolean isSyntheticWidgetTestGroup(@NotNull Group group) {
-    return Objects.equals(group.getUrl(), "package:flutter_test/src/widget_tester.dart") &&
-           Objects.equals(group.getName(), SYNTHETIC_WIDGET_GROUP_NAME);
+  private static boolean isSyntheticWidgetTestGroup(@Nullable Item group) {
```
Comment (devoncarew) [inline src/io/flutter/run/test/FlutterTestEventsConverter.java]: rename the param to `item`?

File: src/io/flutter/run/test/FlutterTestEventsConverter.java [hunk 2]
```diff
@@ -35,9 +35,9 @@ public FlutterTestEventsConverter(@NotNull String testFrameworkName,
...
+    return group instanceof Group && Objects.equals(group.getUrl(), "package:flutter_test/src/widget_tester.dart") &&
+           group.getName().endsWith(SYNTHETIC_WIDGET_GROUP_NAME);
```
Comment (devoncarew) [inline src/io/flutter/run/test/FlutterTestEventsConverter.java]: I'm guessing group.getName() will always be non-null?

Comment (pq) [inline src/io/flutter/run/test/FlutterTestEventsConverter.java]: Well, I *think* so but I'm not sure it's specified so better safe than sorry; will fix.

Thanks!

File: src/io/flutter/run/test/FlutterTestEventsConverter.java (hunk 1 above)
Comment (pq) [inline src/io/flutter/run/test/FlutterTestEventsConverter.java]: Sure.  Done.

**Comment by pq** [comment]
https://github.com/flutter/flutter-intellij/issues/1260 opened to track upstreaming 👍

**Pull request merged**